src/components/Comment/index.tsx

```diff
@@ -305,10 +362,10 @@ export class CommentBase extends React.Component<Props, State> {
   }
 
   render() {
-    const { className, readOnly } = this.props;
+    const { beginEdit, className, readOnly } = this.props;
     return (
       <div className={makeClassName(styles.container, className)}>
-        {readOnly ? this.renderComment() : this.renderForm()}
+        {readOnly && !beginEdit ? this.renderComment() : this.renderForm()}
```

Comment:
This change makes it so the read-only mode is controlled in two places. First, in the [comment list](https://github.com/mozilla/addons-code-manager/blob/5c957fe5cc75afd87d8102b9d966c159c6443e38/src/components/CommentList/index.tsx#L85-L98) and secondly here in a state-mapped prop. This is a recipe for future bugs since it will be hard to keep both controllers in sync. 

I suggest getting rid of the read-only prop and rely solely on state (which seems to work).

<details>
<summary>Patch</summary>

```diff
diff --git a/src/components/Comment/index.tsx b/src/components/Comment/index.tsx
index 26f5c9d..47ad9f4 100644
--- a/src/components/Comment/index.tsx
+++ b/src/components/Comment/index.tsx
@@ -37,11 +37,11 @@ export type DefaultProps = {
 };
 
 type PropsFromState = {
-  beginEdit: boolean;
   considerDiscard: boolean;
   initialComment: Comment | null;
   initialCommentText: string | null;
   savingComment: boolean;
+  showForm: boolean;
 };
 
 type Props = PublicProps & DefaultProps & PropsFromState & ConnectedReduxProps;
@@ -362,10 +362,10 @@ export class CommentBase extends React.Component<Props, State> {
   }
 
   render() {
-    const { beginEdit, className, readOnly } = this.props;
+    const { className, showForm } = this.props;
     return (
       <div className={makeClassName(styles.container, className)}>
-        {readOnly && !beginEdit ? this.renderComment() : this.renderForm()}
+        {showForm ? this.renderForm() : this.renderComment()}
       </div>
     );
   }
@@ -391,7 +391,7 @@ const mapStateToProps = (
     versionId,
   });
   return {
-    beginEdit: info ? info.beginNewComment : false,
+    showForm: info ? info.beginNewComment : false,
     considerDiscard: info ? info.considerDiscard : false,
     initialComment: initialComment || null,
     initialCommentText: info ? info.pendingCommentText : null,

```

</details>


However, I'd say try to tackle this only at the end. If it makes the patch too big then let's defer it to a future issue.

Reply:
Filed #1286